samples/msal-browser-samples/BrokerTestApp2.0/app/brokeredWidget/script.js

```diff
@@ -0,0 +1,130 @@
+
+// Config object to be passed to Msal on creation
+const msalConfig = {
+    auth: {
+        clientId: "abdd063b-76df-4d97-afc3-05dd10c8b017"
+    },
+    cache: {
+        cacheLocation: "sessionStorage", // This configures where your cache will be stored
+        storeAuthStateInCookie: false, // Set this to "true" if you are having issues on IE11 or Edge
+    },
+    system: {
+        loggerOptions: {
+            loggerCallback: (level, message, containsPii) => {
+                if (containsPii) {
+                    return;
+                }
+                switch (level) {
+                    case msal.LogLevel.Error:
+                        console.error(message);
+                        return;
+                    case msal.LogLevel.Info:
+                        console.info(message);
+                        return;
+                    case msal.LogLevel.Verbose:
+                        console.debug(message);
+                        return;
+                    case msal.LogLevel.Warning:
+                        console.warn(message);
+                        return;
+                }
+            }
+        }
+    },
+    experimental: {
+        brokerOptions: {
+            allowBrokering: true,
+            trustedBrokerDomains: ["http://localhost:30663"],
+            brokeredRedirectUri: "http://localhost:30662"
+        }
+    }
+};
+
+let username;
+
+function handleResponse(resp) {
+    if (resp !== null) {
+        username = resp.account.username;
+    } else {
+        // need to call getAccount here?
+        const currentAccounts = myMSALObj.getAllAccounts();
+        if (!currentAccounts || currentAccounts.length < 1) {
+            return;
+        } else if (currentAccounts.length > 1) {
+            // Add choose account code here
+        } else if (currentAccounts.length === 1) {
+            username = currentAccounts[0].username;
+        }
+    }
+}
+
+const myMSALObj = new msal.PublicClientApplication(msalConfig);
+
+
+myMSALObj.initializeBrokering().then(() => {
+        // Must ensure that initialize has completed before calling any other MSAL functions
+    myMSALObj.handleRedirectPromise().then(handleResponse).catch(err => {
+        console.error(err);
+    });
+
+
+setTimeout(() => {
+    const contentElement = document.getElementsByClassName("myContent")[0];
+    contentElement.innerHTML = "I am now trying to fetch profile data";
+
+    setTimeout(async () => {
+        let exit = false;
+        await myMSALObj.ssoSilent({ loginHint: "dathayil@microsoft.com" })
```

Comment:
Probably should remove hardcoded email?

Reply:
Removed in a later PR, but will also remove it here.